### ISSUE TITLE
Not able to launch dashboard after one of the existing master node is powered down.

### ISSUE THREAD

**Issue by sangeethah** [rancher/rke#37]
rke version v0.0.3-dev

Steps to reproduce the problem:
Create a cluster with 1 node in controlplane , 1 in etcd and 1 in worker.
Also have dashboard installed as addons.
Add 1 more node in controlplane and 1 more node in woker.
Make sure we are able to launch bashboard successfully.

Power off one of the nodes in the controlplane which has the bashboard container running.

After sometime, dashboard gets launched in the new controlplane node. But is not able to launch correctly.

```
root@sangee-now-rke-4:~# docker logs bc27c
2017/11/22 21:25:10 Starting overwatch
2017/11/22 21:25:10 Using in-cluster config to connect to apiserver
2017/11/22 21:25:10 Using service account token for csrf signing
2017/11/22 21:25:10 No request provided. Skipping authorization
2017/11/22 21:25:10 Error while initializing connection to Kubernetes apiserver. This most likely means that the cluster is misconfigured (e.g., it has invalid apiserver certificates or service accounts configuration) or the --apiserver-host param points to a server that does not exist. Reason: the server has asked for the client to provide credentials
Refer to the troubleshooting guide for more information: https://github.com/kubernetes/dashboard/blob/master/docs/user-guide/troubleshooting.md
```

**Comment by galal-hussein** [comment]
@sangeethah In DO hostnames are not resolvable so kubeapi will not be able to reach the node by hostname, it will cause exec/logs to fail, to solve this you can just specify the following extra arguments for kubeapi service:
```
extra_args:
  kubelet-preferred-address-types: "InternalIP,ExternalIP,Hostname"
```

**Comment by moelsayed** [comment]
Currently, when a new node is added to the cluster, a new KubeAPI cert/key pair is generated for it. 

The same key is used to sign service account tokens. The results is an auth split brain where tokens signed by one API server can't authenticate with the other API server.

**Comment by galal-hussein** [comment]
The fix can be tested with v0.0.5-dev

**Comment by sangeethah** [comment]
rke version v0.0.5-dev

Not able to reproduce the issue reported in the bug when following steps were done:

Create a cluster with 1 node in controlplane , 1 in etcd and 1 in worker.
Also have dashboard installed as addons.
Change existing node in etcd to be in both controlplane and etcd.
Add 1 more node in controlplane and 1 more node in woker.
Make sure we are able to launch bashboard successfully.

Power off one of the nodes in the controlplane which has the bashboard container running.
After sometime, dashboard gets launched in the new controlplane node and is able to launch successfully.


Also tested with the following commands to make sure , pod containers deployed in existing and new worker planes is able to reach api servers that were existing and new using the service account tokens:

```
KUBE_TOKEN=$(</var/run/secrets/kubernetes.io/serviceaccount/token)
curl -sSk -H "Authorization: Bearer $KUBE_TOKEN"  https://<controlplane node ip>:6443/api/v1/namespaces/default/pods/$HOSTNAME
```
